### DISCUSSION TITLE
Displaying the types in a column in the meeting list view.

### DISCUSSION THREAD

**Discussion by steve970** [code4recovery/12-step-meeting-list#1531] [Q&A] [answered]
I am trying to get the types to show up as a column in my meeting list view. I am using the TSML UI. Is there a way to go about doing that? Thanks!

**Comment by joshreisner** [comment]
not as its own column, no

there is the ability to customize which 'flags' show up (these are the little "Men" and "Women" that show up next to the meeting name. this should be used if there is a specific critical type that would be necessary for some users to see before tapping on the meeting to learn more about it

```
    <script>
      var tsml_react_config = {
        'flags': ['M', 'W', 'O', 'C'],
      };
    </script>
```

**Comment by steve970** [comment]
Great, good to know. I'm curious if I am able to get any of the other columns to appear on the meeting list view like 'Sub Region' or 'District'?

**Reply by joshreisner** [reply, marked as answer]
the region column should already be displaying the sub-region, if present. (it shows the bottom-most region in the stack)

but these are the only columns we support at the moment

**Reply by waiwebsitecommittee-lab** [reply]
I would also like to add this column of data to the display in place of the Region column (which is unhelpful in our small area).  Where would I need to go in the code to work on changing that?  Can you specify which file in the repo configures the display to show region?

**Comment by steve970** [comment]
When I insert this above code snippet, I only see one type show up as a 'flag'. Is there away to get all of the types I put into the snippet to show up? Thanks!

**Reply by joshreisner** [reply]
could you provide a link to your site?

**Reply by steve970** [reply]
Unfortunately, I am running locally and am not able to provide a link.  I can only see Men and Women flags when I put the script in, but if I tag a meeting with say Men and Open, the only type that shows up is Men.

**Reply by joshreisner** [reply]
hard to know what's going on - i don't think there's a bug where it will only display one. i see a few multiples in this example: https://tsml-ui.code4recovery.org/tests/aasepia.html

<img width="523" alt="Screenshot 2024-09-30 at 3 09 30 PM" src="https://github.com/user-attachments/assets/12987010-5b6c-4874-a76a-11c36759ae2a">

i'd double check that types are in the JSON and that the codes match exactly?

i would match the data by looking at the "slug" of the meeting like htis:
<img width="1920" alt="Untitled 2" src="https://github.com/user-attachments/assets/f10901df-b485-447d-81d8-5cf5d3c8023d">

and the code we used is here:
https://github.com/code4recovery/tsml-ui/blob/main/public/tests/aasepia.html#L32

**Reply by steve970** [reply]
In this script: 

<script>
      var tsml_react_config = {
        'flags': ['M', 'W', 'O', 'C'],
      };
</script>

The flags you provided were Men, Women, Open, Close.  When I add that code snippet, the only flag that actually populates is Women. It's the same on your demo. What makes a flag visible vs. not visible on the meeting list view.

In my head, let's say I filtered the demo meeting list by LGBTQ. I'd expect to see flags for open and closed meetings without having to go through each individual meeting to find which is which. Does that make sense?

**Reply by joshreisner** [reply]
yes, that's how it should work currently. if you can post your data perhaps we can spot where the problem is?

one thing to note is that these specific types are special, so meetings that are tagged Women are necessarily not tagged Men, and meetings that are typed Closed are necessarily not tagged Open. But if you have a meeting with `types: ["C", "W"]` i would expect to see `Closed, Women` displayed after the meeting name, regardless of what the current filter was

**Reply by joshreisner** [reply]
oh i think i know what the issue is, you are using TSML and it is overwriting the TSML UI  configuration with its own. sorry, i'm not sure why i was thinking you were running TSML UI by itself. 

in this case forget the javascript - you'll need to add some PHP code to your theme's functions.php instead:

```
if (function_exists('tsml_custom_flags')) {
    tsml_custom_flags(array('M', 'W', 'O', 'C'));
}
```

**Reply by steve970** [reply]
That worked great. Thank you for your time, help, and service with my question.
